### PR TITLE
feat: add share table query

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. feat: add share table query
2. add `share_endpoint_address` config, if it is empty, create a dummy operator.

Fixes https://github.com/datafuselabs/databend/issues/7820

### CODE REVIEW

File: src/common/storage/src/operator.rs [hunk 1]
```diff
@@ -255,6 +255,7 @@ fn init_obs_operator(cfg: &StorageObsConfig) -> Result<Operator> {
 #[derive(Clone, Debug)]
 pub struct StorageOperator {
     operator: Operator,
+    share_endpoint_address: Option<String>,
```
Comment (Xuanwo) [inline src/common/storage/src/operator.rs:258]: We should not touch this struct.

File: src/common/storage/src/operator.rs [hunk 2]
```diff
@@ -271,20 +272,25 @@ static STORAGE_OPERATOR: OnceCell<Singleton<StorageOperator>> = OnceCell::new();
 impl StorageOperator {
     pub async fn init(
         conf: &StorageConfig,
+        share_endpoint_address: &str,
         v: Singleton<StorageOperator>,
     ) -> common_exception::Result<()> {
-        v.init(Self::try_create(conf).await?)?;
+        v.init(Self::try_create(conf, share_endpoint_address).await?)?;
```
Comment (Xuanwo) [inline src/common/storage/src/operator.rs:278]: same here.

File: src/query/sharing/src/layer.rs [hunk 3]
```diff
@@ -109,4 +142,23 @@ impl Accessor for SharedAccessor {
         let resp = self.client.send_async(req).await?;
         Ok(resp.into_body().reader())
     }
+
+    async fn stat(&self, path: &str, args: OpStat) -> Result<ObjectMetadata> {
+        let (_, _) = (path, args);
+        Ok(ObjectMetadata::new(ObjectMode::FILE))
```
Comment (Xuanwo) [inline src/query/sharing/src/layer.rs:148]: We don't need metadata? What happens if this file does not exist?

File: src/common/storage/src/operator.rs (hunk 1 above)
Comment (lichuang) [reply src/common/storage/src/operator.rs:258]: we need some way to get the share endpoint address when creating FuseTable, where will be more suitable?

Comment (Xuanwo) [reply src/common/storage/src/operator.rs:258]: How about placing it in table options?

Comment (lichuang) [reply src/common/storage/src/operator.rs:258]: `TableInfo` is return by meta server, and meta is unaware of share endpoint.

Comment (Xuanwo) [reply src/common/storage/src/operator.rs:258]: I think we need to refactor the `get_table_info` API and insert `endpoint_address` there.

File: src/query/storages/fuse/src/fuse_table.rs [hunk 4]
```diff
@@ -157,13 +172,28 @@ impl FuseTable {
...
     }
 
-    pub fn snapshot_loc(&self) -> Option<String> {
-        let options = self.table_info.options();
-        options
-            .get(OPT_KEY_SNAPSHOT_LOCATION)
-            // for backward compatibility, we check the legacy table option
-            .or_else(|| options.get(OPT_KEY_LEGACY_SNAPSHOT_LOC))
-            .cloned()
+    pub async fn snapshot_loc(&self) -> Option<String> {
```
Comment (BohuTANG) [inline src/query/storages/fuse/src/fuse_table.rs:175]: It's better to return `Result<Option<String>>`

**Comment by BohuTANG** [comment]
LGTM, cc @dantengsky for final review :)

File: src/common/exception/src/exception_code.rs [hunk 5]
```diff
@@ -145,6 +145,7 @@ build_exceptions! {
     InvalidTimestamp(1080),
     InvalidClusterKeys(1081),
     UnknownFragmentExchange(1082),
+    InvalidUtf8(1083),
```
Comment (Xuanwo) [inline src/common/exception/src/exception_code.rs:148]: I don't think it's a meaning error code that should be returned to users. How about use `InternalError`?

**Comment by dantengsky** [comment]
@lichuang sorry for the late response, being trapped in the bug-fixing task.

**Review by dantengsky — APPROVED** [review]
LGTM

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**